Backpack/src/main/java/net/skyscanner/backpack/util/TextLayoutCompat.kt

```diff
@@ -0,0 +1,182 @@
+/**
+ * Backpack for Android - Skyscanner's Design System
+ *
+ * Copyright 2018-2020 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.skyscanner.backpack.util
+
+import android.os.Build
+import android.text.Layout
+import android.text.StaticLayout
+import android.text.TextDirectionHeuristic
+import android.text.TextDirectionHeuristics
+import android.text.TextPaint
+import android.text.TextUtils
+import android.util.Log
+import androidx.annotation.RequiresApi
+import java.lang.reflect.Constructor
+import net.skyscanner.backpack.BuildConfig
+
+internal object TextLayoutCompat {
+
+  private const val TAG = "TextLayoutCompat"
+
+  private val creator = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
```

Comment:
Prior to Android API 23 (Marshmallow) Android doesn't provide any public API to make static layout multi-line.